src/vm/reflectioninvocation.cpp

```diff
@@ -2857,6 +2862,7 @@ FCIMPL3(void, ReflectionInvocation::ExecuteCodeWithGuaranteedCleanup, Object* co
         // attempt to run the backout code.
         PrepareMethodDesc(g_pExecuteBackoutCodeHelperMethod, Instantiation(), Instantiation(), FALSE, TRUE);
     }
+#endif // FEATURE_CER
 
     ExecuteCodeWithGuaranteedCleanupHelper(&gc);
```

Comment:
We should implement the whole ExecuteCodeWithGuaranteedCleanup in C#, have try/catch around the cleanup code, and call Environment.FailFast if there is any exception during the cleanup.